packages/babel-plugin-transform-es2015-block-scoping/test/fixtures/general/switch/actual.js

```diff
@@ -9,3 +11,15 @@ switch (a) {
     let d = 3;
     break;
 }
+
+switch (true) {
+  case true:
+    let a = false;
+    let b = true;
+    let c = 4;
+    let d = 5;
+
+  case false:
+    class e {}
+    function f() {}
+}
```

Comment:
It doesn't currently, I'll add it. 👍

to be honest, there are more edge cases not currently covered with lexical scope in general but IMO that's outside the scope of this PR since I'm just trying to fix this bug 😏